docs/gitlab_integration.md

```diff
@@ -0,0 +1,132 @@
+# 1. Install/register GitLab Runner
+GitLab Runner is the open source project that is used to run your jobs and
+send the results back to GitLab. You have to
+[install](https://docs.gitlab.com/runner/install/) it on a machine where
+your analysis will be performed.
+
+Once GitLab Runner is installed, you need to register it with GitLab.
+You can use the following command:
+```sh
+sudo gitlab-runner register \
+    --non-interactive \
+    --url "https://mycompany.gitlab.com" \
+    --registration-token "$REGISTRATION_TOKEN" \
+    --description "codechecker" \
+    --executor "shell"
+```
+
+The token can be found in your projects settings under
+`settings/ci_cd/Set up a specific Runner manually`. You can find more
+information how to
+[register a GitLab Runner](https://docs.gitlab.com/runner/register/index.html).
+
+# 2. Define a job in your .gitlab-ci.yml
+GitLab CI/CD pipelines are configured using a YAML file called `.gitlab-ci.yml`
+within each project. Create this file in your project root directory:
+
+```yml
+code_quality:
+  allow_failure: false
+  script:
+    - bash .gitlab/ci/run_codechecker.sh
+  after_script:
+    - cat gl-code-quality-report.json
+  artifacts:
+    reports:
+      codequality: gl-code-quality-report.json
+    paths: [gl-code-quality-report.json]
+    expire_in: '5 yrs'
+  stage: test
+```
+
+`expire_in` allows you to specify how long artifacts should live before they
+expire and are therefore deleted. You can set it to a lower or a higher value.
+For more information
+[see](https://docs.gitlab.com/ee/ci/yaml/README.html#artifactsexpire_in).
+
+Script is a shell script which will be executed by the Runner.
+
+# 3. Create script to run CodeChecker analysis
+Create a shell script in your project under `.gitlab/ci/run_codechecker.sh`.
+This script will run the analysis and it has to create a json file called
+`gl-code-quality-report.json` which will contains reports in Code Climate
+format. CodeChecker from version `6.12.0` is able to generate Code Climate
+output by using the *CodeChecker parse* or *CodeChecker cmd diff* commands.
```

Comment:
I'm not sure what is the best solution, should we list all the sub commands which have this output format?
If yes we will need to update this list if the codeclimate format is introduced for any other command later.
My concern is that we will probably forgot it.
But if it is not listed here probably it will be harder to find.